tools/nnpackage_tool/model2nnpkg/model2nnpkg

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
```

Comment:
Unfortunately, we cannot give executable permission to files in this repo. The format checker may block executable files in here.
https://github.com/Samsung/ONE/blob/71c30e1a3e3b1aaed09c0f9772118e8e6859d0b2/infra/command/format#L83-L99

Reply:
@ragmani By default, our format checker does not allow executable mode except for `*.py` and `*.sh`. But there are several exceptions (`nnfw`, `nnas`, `infra/debian/compiler/rules`, ...). We may add `model2nnpkg` in this allowed list.

cc @hseok-oh

Reply:
@hseok-oh  As I remember, the executable mask check was introduced to prevent source codes (which are not executables) with executable permission, mainly from Windows developer by mistake. Then, we may change `format` check rule.

#### As-is
Maintain allow-list as the code above.

#### Suggestion
Check using `file` command.
```sh
$ file <FILES_TO_CHECK_PERMISSION>
infra/debian/compiler/rules:     a /usr/bin/make -f script, ASCII text executable
infra/debian/runtime/rules:      a /usr/bin/make -f script, ASCII text executable
nnas:                            Bourne-Again shell script, ASCII text executable
nncc:                            Bourne-Again shell script, ASCII text executable
nnfw:                            Bourne-Again shell script, ASCII text executable
runtime/contrib/android/gradlew: a /usr/bin/env sh script, ASCII text executable
```

Reply:
We are allowing executable permission for

- Command `nnas`, `nnfw`, `nncc` 
  - We had plan to merge one command (`nnas` when we planned) when we merged two repo, but it stopped now
- `/rules`
  - For debian package
- `/gradlew`
  - For android package
- `*.sh` and `*.py`
  - Used on CI
  - Used as host tool without install

At first, there was only one exception (`nnfw`)  for command, but policy added because packaging tools require permission. I don't want add more exception.

Reply:
@hseok-oh 

1.

> We are allowing executable permission for ...

May I ask why? Why we don't allow executable permission for executable?

Is it related to some vulnerability? If so, is it safe to allow exceptions?

2.

>  I don't want to add more exception.

May I ask why? To reduce maintenance effort ?

Then, what do you think about https://github.com/Samsung/ONE/pull/10221#discussion_r1054040584 ?

3. If you want to keep the rule (which I don't understand why), it means

We will change UX if similar things happen (implementation language changes) for our tools. (Not only `model2nnpkg`, but for all tools), we will break the backward compatibility. Users code will not work.

Reply:
>May I ask why? Why we don't allow executable permission for executable?

Executable on our repo are bash script (`*.sh`) and python script (`*.py`). They are allowed and fomat checker can find by file extension. 
Exception case is commands without file extension `nnas`, `nnfw`, and `nncc`.

>May I ask why? To reduce maintenance effort ?

Right.

>We will change UX if similar things happen (implementation language changes) for our tools. (Not only `model2nnpkg`, but for all tools), we will break the backward compatibility. Users code will not work.

I don't think we should maintain backward compatibility for tools. They are not in our release bound. `model2nnpkg` is used on compiler's `one-pack` internally and compiler package includes `model2nnpkg`, but its interface is not in official support. I think we need to migrate carefully because this tools is used in compiler's official command's implementation, but it is independent issue with maintaining backward compatibility.

Reply:
@hseok-oh 

> > May I ask why? Why we don't allow executable permission for executable?
> 
> Executable on our repo are bash script (`*.sh`) and python script (`*.py`). They are allowed and fomat checker can find by file extension. Exceptional case (executable wthout file extension) are `nnas`, `nnfw`, and `nncc`.

My question is about why, not about what is allowed. : )
I am curious why we don't allow. I wondered whether there is some vulnerability.

> I don't think we should maintain backward compatibility for tools. 

Okay. `model2nnpkg` is internal tool, not official tool for users. Then, just adding `.py` (that is, `model2nnpkg.py`) may be enough.

Reply:
>>May I ask why? Why we don't allow executable permission for executable?

>My question is about why, not about what is allowed. : )

Sorry for Korean. 

실행 권한을 허용하지 않는 이유에 대해서 질문하셔서, 개발자 편의를 위해서 빌드나 설치 과정이 별도로 필요없는 스크립트들은 확장자로 구분할 수 있으면 허용한다고 답변드린 것입니다.
기본적으로 허용하지 않는 이유를 물으신다면, 저는 반대로 source code repository 에서 실행 권한이 있는 file 을 관리해야 할 이유가 빌드 시스템상의 필요성 이외에 무엇인지 묻고 싶습니다. 여기가 실행파일이나 패키지 배포 repository는 아니지 않습니까? 

실행권한이 필요한 script 에 적절한 확장자를 붙이는 것은 예전부터 유지되어온 정책이고, compiler 와 runtime repo 가 합쳐질때 한번 더 확인했었고, 필요한 경우 이름 바꾸는것도 진행했었습니다.

https://github.sec.samsung.net/STAR/nnfw/issues/5747
https://github.sec.samsung.net/STAR/nnfw/pull/5801

>Okay. `model2nnpkg` is internal tool, not official tool for users. Then, just adding `.py` (that is, `model2nnpkg.py`) may be enough.

Agree

Reply:
@hseok-oh  Thank you for explanation. 

> 실행 권한을 허용하지 않는 이유에 대해서 질문하셔서, 개발자 편의를 위해서 빌드나 설치 과정이 별도로 필요없는 스크립트들은 확장자로 구분할 수 있으면 허용한다고 답변드린 것입니다.
기본적으로 허용하지 않는 이유를 물으신다면, 저는 반대로 source code repository 에서 실행 권한이 있는 file 을 관리해야 할 이유가 빌드 시스템상의 필요성 이외에 무엇인지 묻고 싶습니다. 여기가 실행파일이나 패키지 배포 repository는 아니지 않습니까?

I think git source is mainly for developers. Users usually uses the release package only. I thought the repository also need to consider developers.

As I understand,

- executable permission itself does not bring vulnerability
- It does not add more maintenance effort  with https://github.com/Samsung/ONE/pull/10221#discussion_r1054040584
- developers or CI script or install script need to add permission to execute some tool, which is additional cost in my view

> 실행권한이 필요한 script 에 적절한 확장자를 붙이는 것은 예전부터 유지되어온 정책이고, compiler 와 runtime repo 가 합쳐질때 한번 더 확인했었고, 필요한 경우 이름 바꾸는것도 진행했었습니다.
>
> https://github.sec.samsung.net/STAR/nnfw/issues/5747

I didn't know this issue. Thank you for referencing it. There were several items and I don't find any feedback or opinion about permission policy. Also, I think the policy can be updated. The issue above was 3.5 years ago.

. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 

I don't think we need to spend more time on this. `model2nnpkg` itself is not a popular user tool and the change occurs seldom.

Reply:
I did not fully understand the conclusion. Is it correct to add `.py` to python scripts and to give those scripts executable permission?

Reply:
>Is it correct to add .py to python scripts and to give those scripts executable permission?

Right.